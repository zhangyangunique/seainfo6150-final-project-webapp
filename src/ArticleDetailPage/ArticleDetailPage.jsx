import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import styles from "./ArticleDetailPage.module.css";
import Comment from "./Comment/Comment.jsx";
import { isEmpty } from "lodash";

const ArticleDetailPage = (props) => {
	const [fetchedArticle, setFetchedArticle] = useState();

	useEffect(() => {
		const fetchData = async () => {
			// performs a GET request
			const response = await fetch("https://run.mocky.io/v3/ad567970-3ec9-49e0-9750-2a7d9934c2b9");
			const responseJson = await response.json();
			const articles = Object.values(responseJson);

			for (let i = 0; i < articles.length; i++) {
				if (articles[i].slug === props.slug) {
					setFetchedArticle(articles[i]);
				}
			}
		};

		if (isEmpty(fetchedArticle)) {
			fetchData();
		}

	}, [fetchedArticle]);

	return isEmpty(fetchedArticle) ? null : (
		<div className={styles.container}>
			<div className={styles.articleContainer}>
				<article className={styles.article}>
					<div className={styles.imageArea}>
						<img className={styles.image} src={fetchedArticle.image._url} alt={fetchedArticle.title} />
					</div>
					<div className={styles.wrapper}>
						<h2 className={styles.title}>{fetchedArticle.title}</h2>
						<div className={styles.authorinfo}>
							<div className={styles.author}>{fetchedArticle.author}</div>
							<time>{fetchedArticle.pubDate}</time>
						</div>
						<div className={styles.text}>
							{fetchedArticle.text}
						</div>
					</div>
				</article>
			</div>
			<Comment />
		</div>
	);
};

ArticleDetailPage.propTypes = {
	slug: PropTypes.string.isRequired
}

export default ArticleDetailPage;
