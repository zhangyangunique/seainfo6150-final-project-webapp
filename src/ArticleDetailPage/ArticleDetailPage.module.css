.container {
	margin-bottom: 80px;
}

.articleContainer {
	font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
	font-size: 1.5rem;
	margin: 50px 50px;
}

.title {
	text-align: center;
	margin-bottom: 0;
	font-size: 45px;
}

.image {
	width: 100%;
	margin: 0 auto;
	display: block;
}

.author {
	display: inline;
	margin-right: 50px;
}

.authorinfo {
	text-align: center;
	font-size: 26px;
	color: grey;
	margin: 10px;
}

.text {
	font-size: 33px;
}

@media screen and (min-width: 525px) {
	.image {
		width: 90%;
	}
	.title {
		font-size: 40px;
	}
	.authorinfo {
		font-size: 25px;
	}
	.text {
		font-size: 32px;
	}
}

@media screen and (min-width: 800px) {
	.image {
		width: 80%;
	}
	.title {
		font-size: 35px;
	}
	.authorinfo {
		font-size: 20px;
	}
	.text {
		font-size: 28px;
	}
}

@media screen and (min-width: 1150px) {
	.image {
		width: 70%;
	}
	.title {
		font-size: 40px;
	}
	.authorinfo {
		font-size: 18px;
	}
	.text {
		font-size: 25px;
	}
}