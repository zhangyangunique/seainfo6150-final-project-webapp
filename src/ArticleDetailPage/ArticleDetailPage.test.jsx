import React from "react";
import { render } from "@testing-library/react";
import ArticleDetailPage from "./ArticleDetailPage";

const slug = "181122102925-statue-of-liberty-crown";

describe("ArticleDetailPage tests", () => {
    it("renders correctly", () => {
        const { container } = render(<ArticleDetailPage slug={slug} />);
        expect(container).toMatchSnapshot();
    });
});