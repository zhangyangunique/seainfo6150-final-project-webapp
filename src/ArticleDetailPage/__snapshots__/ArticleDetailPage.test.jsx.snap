// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetailPage tests renders correctly 1`] = `<div />`;
